beacon_node/src/cli.rs

```diff
@@ -236,76 +269,111 @@ pub fn cli_app<'a, 'b>() -> App<'a, 'b> {
                       discovery. Set this only if you are sure other nodes can connect to your \
                       local node on this address. This will update the `ip4` or `ip6` ENR fields \
                       accordingly. To update both, set this flag twice with the different values.")
-                .multiple(true)
-                .max_values(2)
-                .takes_value(true),
+                .action(ArgAction::Append)
+                .num_args(0..=2)
+                .display_order(0)
         )
         .arg(
-            Arg::with_name("enr-match")
-                .short("e")
+            Arg::new("enr-match")
+                .short('e')
                 .long("enr-match")
+                .action(ArgAction::SetTrue)
+                .help_heading(FLAG_HEADER)
                 .help("Sets the local ENR IP address and port to match those set for lighthouse. \
                       Specifically, the IP address will be the value of --listen-address and the \
                       UDP port will be --discovery-port.")
+                .display_order(0)
         )
         .arg(
-            Arg::with_name("disable-enr-auto-update")
-                .short("x")
+            Arg::new("disable-enr-auto-update")
+                .short('x')
```

Comment:
Not sure why the short name here is `x`? This flag appears just before `zero-ports` which is seperated from other `--disable` flags. Current flags although with x, but it still shows in alphabetical order starting with `d`

Reply:
i ended up removing the short name